src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java

```diff
@@ -198,14 +208,27 @@ else if (len < 2000)
 
         if (ENABLE_BOOKKEEPING)
         {
-            logger.info("Thread " + threadId() + " returned array "
-                    + System.identityHashCode(buf));
             Integer arrayId = System.identityHashCode(buf);
-            if (bookkeeping.remove(arrayId) == null)
+            logger.info("Thread " + threadId() + " returned array "
+                    + arrayId);
+            if (bookkeeping.containsKey(arrayId))
+            {
+                returnedBookkeeping.put(arrayId, new ReturnedBufferBookkeepingInfo(bookkeeping.get(arrayId), UtilKt.getStackTrace()));
+                bookkeeping.remove(arrayId);
+            }
+            else if (returnedBookkeeping.containsKey(arrayId))
+            {
+                logger.info("Thread " + threadId()
+                    + " returned a previously-returned buffer at\n"
+                    + UtilKt.getStackTrace() +
+                    "previously returned at\n" +
+                    returnedBookkeeping.get(arrayId).deallocTrace);
```

Comment:
I've seen a race happen here where this causes an NPE...so we should check and save the entry in one step.